File: app/models/membership.py
from app.extensions import db
from sqlalchemy.sql import func

class Membership(db.Model):
    __tablename__ = 'membership'

    membership_id = db.Column(db.BigInteger, primary_key=True, autoincrement=True)
    membership_customer_id = db.Column(db.String(100), nullable=False)
    membership_sub_id = db.Column(db.String(100), nullable=False)
    membership_variant_id = db.Column(db.String(100), nullable=False)  # ex) Premium
    membership_status = db.Column(db.String(20), nullable=False)       # ACTIVE, CANCELLED 등
    membership_period_end = db.Column(db.DateTime, nullable=False)
    membership_start_date = db.Column(db.Date, nullable=False)

    membership_created_at = db.Column(db.TIMESTAMP, server_default=func.now(), nullable=False)
    membership_updated_at = db.Column(db.TIMESTAMP, server_default=func.now(), onupdate=func.now(), nullable=False)

    # 외래키
    user_no = db.Column(db.BigInteger, db.ForeignKey('user.user_no'), nullable=False)

File: src/main/java/com/ticketflow/entity/Membership.java
package com.ticketflow.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "membership")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Membership {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "membership_id")
    private Long membershipId;

    @Column(name = "membership_customer_id", length = 100, nullable = false)
    private String membershipCustomerId;

    @Column(name = "membership_sub_id", length = 100, nullable = false)
    private String membershipSubId;

    @Column(name = "membership_variant_id", length = 100, nullable = false)
    private String membershipVariantId;

    @Column(name = "membership_status", length = 20, nullable = false)
    private String membershipStatus;

    @Column(name = "membership_period_end", nullable = false)
    private LocalDateTime membershipPeriodEnd;

    @Column(name = "membership_start_date", nullable = false)
    private LocalDate membershipStartDate;

    @CreationTimestamp
    @Column(name = "membership_created_at", updatable = false, nullable = false)
    private LocalDateTime membershipCreatedAt;

    @UpdateTimestamp
    @Column(name = "membership_updated_at", nullable = false)
    private LocalDateTime membershipUpdatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_no", nullable = false)
    private User user;
}
